src/block_ffm.rs

```diff
@@ -118,37 +120,88 @@ impl <L:OptimizerTrait + 'static> BlockTrait for BlockFFM<L>
-        if mi.ffm_k > 0 {       
-            if mi.ffm_init_width == 0.0 {
-                // Initialization that has showed to work ok for us, like in ffm.pdf, but centered around zero and further divided by 50
-                let ffm_one_over_k_root = 1.0 / (self.ffm_k as f32).sqrt() / 50.0;
-                for i in 0..self.ffm_weights_len {
-                    self.weights[i as usize].weight = (1.0 * merand48((self.ffm_weights_len as usize+ i as usize) as u64)-0.5) * ffm_one_over_k_root;
-                    self.weights[i as usize].optimizer_data = self.optimizer_ffm.initial_data();
-                }
-            } else {
-                let zero_half_band_width = mi.ffm_init_width * mi.ffm_init_zero_band * 0.5;
-                let band_width = mi.ffm_init_width * (1.0 - mi.ffm_init_zero_band);
-                for i in 0..self.ffm_weights_len {
-                    let mut w = merand48(i as u64) * band_width - band_width * 0.5;
-                    if w > 0.0 { 
-                        w += zero_half_band_width ;
-                    } else {
-                        w -= zero_half_band_width;
-                    }
-                    w += mi.ffm_init_center;
-                    self.weights[i as usize].weight = w; 
-                    self.weights[i as usize].optimizer_data = self.optimizer_ffm.initial_data();
-                }
 
-            }
-        }
+		if mi.initialization_type.eq("default"){
+			if mi.ffm_k > 0 {       
+				if mi.ffm_init_width == 0.0 {
+					// Initialization that has showed to work ok for us, like in ffm.pdf, but centered around zero and further divided by 50
+					let ffm_one_over_k_root = 1.0 / (self.ffm_k as f32).sqrt() / 50.0;
+					for i in 0..self.ffm_weights_len {
+						self.weights[i as usize].weight = (1.0 * merand48((self.ffm_weights_len as usize+ i as usize) as u64)-0.5) * ffm_one_over_k_root;
+						self.weights[i as usize].optimizer_data = self.optimizer_ffm.initial_data();
+					}
+				} else {
+					let zero_half_band_width = mi.ffm_init_width * mi.ffm_init_zero_band * 0.5;
+					let band_width = mi.ffm_init_width * (1.0 - mi.ffm_init_zero_band);
+					for i in 0..self.ffm_weights_len {
+						let mut w = merand48(i as u64) * band_width - band_width * 0.5;
+						if w > 0.0 { 
+							w += zero_half_band_width ;
+						} else {
+							w -= zero_half_band_width;
+						}
+						w += mi.ffm_init_center;
+						self.weights[i as usize].weight = w; 
+						self.weights[i as usize].optimizer_data = self.optimizer_ffm.initial_data();
+					}
+
+				}
+			}
+		} else if mi.initialization_type.eq("xavier") {
+			// U [-(1/sqrt(n)), 1/sqrt(n)]
+
+			let lower_bound: f32 = -1.0/(self.ffm_weights_len as f32).sqrt();
+			let upper_bound: f32 = 1.0/(self.ffm_weights_len as f32).sqrt();
+			let difference = upper_bound - lower_bound;
+			
+			for i in 0..self.ffm_weights_len {
```

Comment:
loop code here and in the next else-if block is exactly the same and is similar to loop in other if-else blocks, please extract common code into functions

Reply:
Actually not 100% sure that's the way to go here -- we are within a trait block, meaning that the added function would need to be added also across the`regressor.rs` and possibly elsewhere, actually increasing the amount of source -- this would make sense if this was something really generic, but I'm not so sure it is (+ it might be easier to maintain this way for now), wdyt?